[GTK][WPE] Enable modern media controls

philn opened the pull request:
#### 5c4b162cb7329833ecab8386e9762313ddc3444c
<pre>
[GTK][WPE] Enable modern media controls
<a href="https://bugs.webkit.org/show_bug.cgi?id=182502">https://bugs.webkit.org/show_bug.cgi?id=182502</a>

Patch by Philippe Normand &lt;philn@igalia.com &gt; on 2022-06-02
Reviewed by Adrian Perez de Castro.

Initial support for the modern media controls on GTK and WPE. The old media controls are broken,
their buttons are no longer rendered.

These new controls are adapted from the macOS controls, both inline and fullscreen, with the
difference that we use the Adwaita icon theme. SVG/PNG icons are shipped in the WebKit library using
GResources. Some adaptations had to be made in the cross-platform JS code of the controls, so that
Airplay UI components can be disabled, WPE/GTK don&apos;t support this yet.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* Source/WTF/wtf/Platform.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::layoutTraitsClassName const):
* Source/WebCore/Modules/mediacontrols/mediaControlsAdwaita.css: Removed.
* Source/WebCore/Modules/mediacontrols/mediaControlsAdwaita.js: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-fullscreen-media-controls.css: Added.
(.media-controls.adwaita.fullscreen):
(.media-controls.adwaita.fullscreen  &gt; .controls-bar):
(.media-controls.adwaita.fullscreen:not(.uses-ltr-user-interface-layout-direction) .volume.slider):
(.media-controls.adwaita.fullscreen .buttons-container):
(.media-controls.adwaita.fullscreen .buttons-container.left):
(.media-controls.adwaita.fullscreen .buttons-container.center):
(.media-controls.adwaita.fullscreen .buttons-container.right):
(.media-controls.adwaita.fullscreen .buttons-container.right button):
(.media-controls.adwaita.fullscreen .time-control):
(.media-controls.adwaita.fullscreen  &gt; .controls-bar .status-label):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-fullscreen-media-controls.js: Added.
(AdwaitaFullscreenMediaControls.prototype.handleEvent):
(AdwaitaFullscreenMediaControls.prototype.layout):
(AdwaitaFullscreenMediaControls.prototype._volumeControlsForCurrentDirection):
(AdwaitaFullscreenMediaControls.prototype._collapsableButtons):
(AdwaitaFullscreenMediaControls.prototype._handleMousedown):
(AdwaitaFullscreenMediaControls.prototype._handleMousemove):
(AdwaitaFullscreenMediaControls.prototype._handleMouseup):
(AdwaitaFullscreenMediaControls.prototype._pointForEvent):
(AdwaitaFullscreenMediaControls):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-inline-media-controls.css: Added.
(.media-controls.adwaita.inline .volume-slider-container):
(.media-controls.adwaita.inline.audio .volume-slider-container):
(.media-controls.adwaita.inline .volume-slider-container  &gt; .background-tint):
(.media-controls.adwaita.inline .volume-slider-container  &gt; .background-tint  &gt; div):
(.media-controls.adwaita.inline .volume-slider-container  &gt; .slider):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-inline-media-controls.js: Added.
(AdwaitaInlineMediaControls.prototype.layout):
(AdwaitaInlineMediaControls.prototype.get preferredMuteButtonStyle):
(AdwaitaInlineMediaControls.prototype.handleEvent):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-layout-traits.js: Copied from Source/WebCore/Modules/modern-media-controls/controls/macos-layout-traits.js.
(AdwaitaLayoutTraits.prototype.mediaControlsClass):
(AdwaitaLayoutTraits.prototype.overridenSupportingObjectClasses):
(AdwaitaLayoutTraits.prototype.resourceDirectory):
(AdwaitaLayoutTraits.prototype.controlsAlwaysAvailable):
(AdwaitaLayoutTraits.prototype.controlsNeverAvailable):
(AdwaitaLayoutTraits.prototype.supportsIconWithFullscreenVariant):
(AdwaitaLayoutTraits.prototype.supportsDurationTimeLabel):
(AdwaitaLayoutTraits.prototype.controlsDependOnPageScaleFactor):
(AdwaitaLayoutTraits.prototype.skipDuration):
(AdwaitaLayoutTraits.prototype.promoteSubMenusWhenShowingMediaControlsContextMenu):
(AdwaitaLayoutTraits.prototype.supportsTouches):
(AdwaitaLayoutTraits.prototype.supportsAirPlay):
(AdwaitaLayoutTraits.prototype.supportsPiP):
(AdwaitaLayoutTraits.prototype.toString):
(AdwaitaLayoutTraits):
* Source/WebCore/Modules/modern-media-controls/controls/inline-media-controls.js:
(InlineMediaControls.prototype._rightContainerButtons):
(InlineMediaControls.prototype._droppableButtons):
* Source/WebCore/Modules/modern-media-controls/controls/ios-layout-traits.js:
(IOSLayoutTraits.prototype.supportsAirPlay):
(IOSLayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/controls/layout-traits.js:
(LayoutTraits.prototype.supportsAirPlay):
(LayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/controls/macos-layout-traits.js:
(MacOSLayoutTraits.prototype.supportsAirPlay):
(MacOSLayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/controls/media-controls.js:
(MediaControls.):
* Source/WebCore/Modules/modern-media-controls/controls/watchos-layout-traits.js:
(WatchOSLayoutTraits.prototype.supportsAirPlay):
(WatchOSLayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/images/adwaita/EnterFullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/ExitFullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Forward.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/MediaSelector-fullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/MediaSelector.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Overflow.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Pause.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/PipIn-fullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/PipIn.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/PipOut.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Play.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Rewind.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipBack10.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipBack15.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipForward10.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipForward15.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume0-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume0.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume1-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume1.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume2-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume2.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume3-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume3.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/VolumeMuted-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/VolumeMuted.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/X.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/invalid-placard@1x.png: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/invalid-placard@2x.png: Added.
* Source/WebCore/Modules/modern-media-controls/media/media-controller.js:
(MediaController.prototype._supportingObjectClasses):
* Source/WebCore/Modules/modern-media-controls/media/media-document-controller.js:
(MediaDocumentController):
* Source/WebCore/PlatformGLib.cmake: Added.
* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::mediaControlsScripts):
(WebCore::RenderThemeAdwaita::mediaControlsStyleSheet):
(WebCore::RenderThemeAdwaita::mediaControlsBase64StringForIconNameAndType):
(WebCore::RenderThemeAdwaita::mediaControlsFormattedStringForDuration):
(WebCore::RenderThemeAdwaita::extraMediaControlsStyleSheet): Deleted.
* Source/WebCore/rendering/RenderThemeAdwaita.h:
* Source/WebKit/ModernMediaControlsGResources.cmake: Added.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/glib/generate-modern-media-controls-gresource-manifest.py: Added.
(get_filenames):

Canonical link: <a href="https://commits.webkit.org/251219@main">https://commits.webkit.org/251219@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@295128">https://svn.webkit.org/repository/webkit/trunk@295128</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

aperezdc commented on Source/WebCore/CMakeLists.txt:
```diff
@@ -2091,6 +2091,121 @@ if (ENABLE_WEBXR)
+add_custom_command(
+    OUTPUT ${WebCore_DERIVED_SOURCES_DIR}/ModernMediaControls.js
+    DEPENDS ${MODERN_MEDIA_CONTROLS_SCRIPTS}
+    COMMAND cat ${MODERN_MEDIA_CONTROLS_SCRIPTS} > ${WebCore_DERIVED_SOURCES_DIR}/ModernMediaControls.js
```
We may want to run the output from this command through `Source/JavaScriptCore/Scripts/jsmin.py` as well.

philn replied:
So I tried this, the minifier renamed the `iconService` constant, breaking everything.

aperezdc commented on Source/WebCore/CMakeLists.txt:
```diff
@@ -2091,6 +2091,121 @@ if (ENABLE_WEBXR)
+add_custom_command(
+    OUTPUT ${WebCore_DERIVED_SOURCES_DIR}/ModernMediaControls.css
+    DEPENDS ${MODERN_MEDIA_CONTROLS_STYLE_SHEETS}
+    COMMAND cat ${MODERN_MEDIA_CONTROLS_STYLE_SHEETS} > ${WebCore_DERIVED_SOURCES_DIR}/ModernMediaControls.css
```
This could be piped through `Source/JavaScriptCore/Scripts/cssmin.py` to reduce the size of the output for bundling.

philn replied:
I tried this, also broke the controls, no error was logged though.

aperezdc commented on Source/WebCore/Modules/modern-media-controls/controls/adwaita-layout-traits.js:
```diff
@@ -0,0 +1,102 @@
+        return false;
+    }
+
+    skipDuration()
+    {
+        return 15;
+    }
```
This has 15 as duration, then the corresponding icon has 30 written on it. Shouldn't they match?

aperezdc replied:
Fixed, I pushed a version of the icons which has “15” written in them 😉

aperezdc reviewed (changes requested):
Thanks a ton for working on this @philn! 💪🏼 

The JS/CSS for the media controls is a bit out of my zone of comfort... from what I understand the changes look fine; but I will let others comment on those.

I have a minor nit about packaging: it would be a good idea to run the concatenated JS/CSS through `jsmin.py`/`cssmin.py`, but as we were not doing it already before, feel free to skip that change and we can consider it for a follow-up patch. The rest of the CMake stuff looks great.

Lastly, I see that there are a few inconsistencies with the icons:

- The skip length is set to 15 in code, then the `SkipBack15.svg` icon has 30 written on it, and `SkipForward15.svg` has 1 written on it.
- The `Pause.svg` icon contains a “play” icon (▶️, should be ⏸️).
- The `Play.svg` icon contains a “pause” icon (⏸️, should be ▶️).

Could you please take a look at the icons?

dcrousso commented on Source/WebCore/Modules/modern-media-controls/controls/layout-traits.js:
```diff
@@ -36,6 +36,11 @@ class LayoutTraits
         return this.mode == LayoutTraits.Mode.Fullscreen;
     }
 
+    isApplePlatform()
```
instead of having this explicitly name "Apple", can we have it be specific to the thing it's actually guarding (i.e. `supportsControlBarOverflow`)?

philn replied:
I replaced this with `supportsPiP()`. Overflow UI is supported now :)

dcrousso commented on Source/WebCore/Modules/modern-media-controls/controls/layout-traits.js:
```diff
@@ -36,6 +36,11 @@ class LayoutTraits
 
+    isApplePlatform()
+    {
+        return true;
```
i don't think we want to assume that all subclasses are Apple (or support control bar overflow), so can we have it `throw` and then add this to all the existing subclasses?

dcrousso commented on Source/WebCore/Modules/modern-media-controls/controls/layout-traits.js:
```diff
@@ -98,6 +103,11 @@ class LayoutTraits
 
+    supportsAirPlay()
+    {
+        return true;
```
ditto (:41)

philn replied:
Done

dcrousso commented on Source/WebCore/Modules/modern-media-controls/controls/media-controls.js:
```diff
@@ -38,14 +38,19 @@ class MediaControls extends LayoutNode
-        this.brightnessButton = new BrightnessButton(this);
+
+        if (this.layoutTraits.isApplePlatform()) {
+            this.overflowButton = new OverflowButton(this);
```
is there actually any harm in always having this exist?  im pretty sure `OverflowSupport` will hide the button if the necessary functionality does not exist

philn replied:
I actually realised I should enable the overflow button. There's no technical reason WPE/GTK can't do this, at least the playback rate popup. So I'll rework this part.

philn replied:
Done.

dcrousso commented on Source/WebCore/Modules/modern-media-controls/controls/media-controls.js:
```diff
@@ -38,14 +38,19 @@ class MediaControls extends LayoutNode
+
+        if (this.layoutTraits.isApplePlatform()) {
+            this.overflowButton = new OverflowButton(this);
+            this.brightnessButton = new BrightnessButton(this);
```
ditto (:48), these buttons by themselves dont really do anything.  they have to be added to some `children` somewhere in order to have any effect

dcrousso commented on Source/WebCore/Modules/modern-media-controls/media/media-document-controller.js:
```diff
@@ -44,6 +44,9 @@ class MediaDocumentController
         if (deviceType == "MacIntel")
             deviceType = mediaController.layoutTraits.supportsTouches() ? "ipad" : "mac";
 
+        if (deviceType.startsWith('Linux'))
```
NIT: `else if`

fred-wang committed:
Add fred-wang github account to contributors.json
https://bugs.webkit.org/show_bug.cgi?id=241219

Reviewed by Tim Nguyen.

* metadata/contributors.json: Add my github account.

Canonical link: https://commits.webkit.org/251218@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295127 268f45cc-cd09-0410-ab3c-d52691b4dbfc

philn committed:
[GTK][WPE] Enable modern media controls
https://bugs.webkit.org/show_bug.cgi?id=182502

Patch by Philippe Normand <philn@igalia.com> on 2022-06-02
Reviewed by Adrian Perez de Castro.

Initial support for the modern media controls on GTK and WPE. The old media controls are broken,
their buttons are no longer rendered.

These new controls are adapted from the macOS controls, both inline and fullscreen, with the
difference that we use the Adwaita icon theme. SVG/PNG icons are shipped in the WebKit library using
GResources. Some adaptations had to be made in the cross-platform JS code of the controls, so that
Airplay UI components can be disabled, WPE/GTK don't support this yet.

* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/gtk/TestExpectations:
* Source/WTF/wtf/Platform.h:
* Source/WebCore/CMakeLists.txt:
* Source/WebCore/Modules/mediacontrols/MediaControlsHost.cpp:
(WebCore::MediaControlsHost::layoutTraitsClassName const):
* Source/WebCore/Modules/mediacontrols/mediaControlsAdwaita.css: Removed.
* Source/WebCore/Modules/mediacontrols/mediaControlsAdwaita.js: Removed.
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-fullscreen-media-controls.css: Added.
(.media-controls.adwaita.fullscreen):
(.media-controls.adwaita.fullscreen > .controls-bar):
(.media-controls.adwaita.fullscreen:not(.uses-ltr-user-interface-layout-direction) .volume.slider):
(.media-controls.adwaita.fullscreen .buttons-container):
(.media-controls.adwaita.fullscreen .buttons-container.left):
(.media-controls.adwaita.fullscreen .buttons-container.center):
(.media-controls.adwaita.fullscreen .buttons-container.right):
(.media-controls.adwaita.fullscreen .buttons-container.right button):
(.media-controls.adwaita.fullscreen .time-control):
(.media-controls.adwaita.fullscreen > .controls-bar .status-label):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-fullscreen-media-controls.js: Added.
(AdwaitaFullscreenMediaControls.prototype.handleEvent):
(AdwaitaFullscreenMediaControls.prototype.layout):
(AdwaitaFullscreenMediaControls.prototype._volumeControlsForCurrentDirection):
(AdwaitaFullscreenMediaControls.prototype._collapsableButtons):
(AdwaitaFullscreenMediaControls.prototype._handleMousedown):
(AdwaitaFullscreenMediaControls.prototype._handleMousemove):
(AdwaitaFullscreenMediaControls.prototype._handleMouseup):
(AdwaitaFullscreenMediaControls.prototype._pointForEvent):
(AdwaitaFullscreenMediaControls):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-inline-media-controls.css: Added.
(.media-controls.adwaita.inline .volume-slider-container):
(.media-controls.adwaita.inline.audio .volume-slider-container):
(.media-controls.adwaita.inline .volume-slider-container > .background-tint):
(.media-controls.adwaita.inline .volume-slider-container > .background-tint > div):
(.media-controls.adwaita.inline .volume-slider-container > .slider):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-inline-media-controls.js: Added.
(AdwaitaInlineMediaControls.prototype.layout):
(AdwaitaInlineMediaControls.prototype.get preferredMuteButtonStyle):
(AdwaitaInlineMediaControls.prototype.handleEvent):
* Source/WebCore/Modules/modern-media-controls/controls/adwaita-layout-traits.js: Copied from Source/WebCore/Modules/modern-media-controls/controls/macos-layout-traits.js.
(AdwaitaLayoutTraits.prototype.mediaControlsClass):
(AdwaitaLayoutTraits.prototype.overridenSupportingObjectClasses):
(AdwaitaLayoutTraits.prototype.resourceDirectory):
(AdwaitaLayoutTraits.prototype.controlsAlwaysAvailable):
(AdwaitaLayoutTraits.prototype.controlsNeverAvailable):
(AdwaitaLayoutTraits.prototype.supportsIconWithFullscreenVariant):
(AdwaitaLayoutTraits.prototype.supportsDurationTimeLabel):
(AdwaitaLayoutTraits.prototype.controlsDependOnPageScaleFactor):
(AdwaitaLayoutTraits.prototype.skipDuration):
(AdwaitaLayoutTraits.prototype.promoteSubMenusWhenShowingMediaControlsContextMenu):
(AdwaitaLayoutTraits.prototype.supportsTouches):
(AdwaitaLayoutTraits.prototype.supportsAirPlay):
(AdwaitaLayoutTraits.prototype.supportsPiP):
(AdwaitaLayoutTraits.prototype.toString):
(AdwaitaLayoutTraits):
* Source/WebCore/Modules/modern-media-controls/controls/inline-media-controls.js:
(InlineMediaControls.prototype._rightContainerButtons):
(InlineMediaControls.prototype._droppableButtons):
* Source/WebCore/Modules/modern-media-controls/controls/ios-layout-traits.js:
(IOSLayoutTraits.prototype.supportsAirPlay):
(IOSLayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/controls/layout-traits.js:
(LayoutTraits.prototype.supportsAirPlay):
(LayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/controls/macos-layout-traits.js:
(MacOSLayoutTraits.prototype.supportsAirPlay):
(MacOSLayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/controls/media-controls.js:
(MediaControls.):
* Source/WebCore/Modules/modern-media-controls/controls/watchos-layout-traits.js:
(WatchOSLayoutTraits.prototype.supportsAirPlay):
(WatchOSLayoutTraits.prototype.supportsPiP):
* Source/WebCore/Modules/modern-media-controls/images/adwaita/EnterFullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/ExitFullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Forward.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/MediaSelector-fullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/MediaSelector.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Overflow.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Pause.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/PipIn-fullscreen.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/PipIn.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/PipOut.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Play.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Rewind.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipBack10.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipBack15.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipForward10.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/SkipForward15.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume0-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume0.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume1-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume1.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume2-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume2.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume3-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/Volume3.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/VolumeMuted-RTL.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/VolumeMuted.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/X.svg: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/invalid-placard@1x.png: Added.
* Source/WebCore/Modules/modern-media-controls/images/adwaita/invalid-placard@2x.png: Added.
* Source/WebCore/Modules/modern-media-controls/media/media-controller.js:
(MediaController.prototype._supportingObjectClasses):
* Source/WebCore/Modules/modern-media-controls/media/media-document-controller.js:
(MediaDocumentController):
* Source/WebCore/PlatformGLib.cmake: Added.
* Source/WebCore/PlatformGTK.cmake:
* Source/WebCore/PlatformMac.cmake:
* Source/WebCore/PlatformWPE.cmake:
* Source/WebCore/rendering/RenderThemeAdwaita.cpp:
(WebCore::RenderThemeAdwaita::mediaControlsScripts):
(WebCore::RenderThemeAdwaita::mediaControlsStyleSheet):
(WebCore::RenderThemeAdwaita::mediaControlsBase64StringForIconNameAndType):
(WebCore::RenderThemeAdwaita::mediaControlsFormattedStringForDuration):
(WebCore::RenderThemeAdwaita::extraMediaControlsStyleSheet): Deleted.
* Source/WebCore/rendering/RenderThemeAdwaita.h:
* Source/WebKit/ModernMediaControlsGResources.cmake: Added.
* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformWPE.cmake:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/WebKitFeatures.cmake:
* Tools/glib/generate-modern-media-controls-gresource-manifest.py: Added.
(get_filenames):

Canonical link: https://commits.webkit.org/251219@main
git-svn-id: https://svn.webkit.org/repository/webkit/trunk@295128 268f45cc-cd09-0410-ab3c-d52691b4dbfc
